dashboard/app/models/lesson.rb

```diff
@@ -258,6 +258,16 @@ def lesson_plan_html_url
     localized_lesson_plan || "#{lesson_plan_base_url}/Teacher"
   end
 
+  def lesson_feedback_url
+    url = "https://studio.code.org/form/teacher_lesson_feedback?survey_data[script_name]=#{script.name}&survey_data[lesson_number]=#{relative_position}&survey_data[lesson_name]=#{localized_name}"
+    url += if script.unit_group
+             "&survey_data[course_name]=#{script.unit_group.localized_title}&survey_data[unit_name]=#{script.localized_title}&survey_data[unit_number]=#{script.unit_group_units&.first&.position}"
+           else
+             "&survey_data[course_name]=#{script.localized_title}"
+           end
+    url
+  end
```

Comment:
some of the values you're passing in here probably need to be escaped.

I would suggest looking around for a query params util that will convert a hash into a query string. in addition to escaping, it should handle all of the ? and & concatenations for you.

if you can't find one, please at least do a web-safe escaping on `localized_name` and `localized_title`.

Reply:
I have escaped `localized_name` and `localized_title`. I spent some time looking around at utils and found `to_query` but the issue is I do not want it to escape the `[]`because then it does not work. So using that helper was an issue.

Reply:
ahh, makes sense. thanks for checking!

nice finds -- CGI.escape looks like the right helper to use now, and `.to_query` looks like the right util to try to use again next time.